test/test_utils.h

```diff
@@ -52,6 +52,8 @@ void mp_msg(struct mp_log *log, int lev, const char *format, ...)
 int mp_msg_find_level(const char *s);
 int mp_msg_level(struct mp_log *log);
 void mp_write_console_ansi(void);
+typedef struct AVDictionary AVDictionary;
```

Comment:
Instead of this, you can use a `#include <libavutil/dict.h>`. We link libavutil to all the tests anyway.

Reply:
It shouldn't. It doesn't for me anyway. Did you put it in `test/test_utils.c`?

Reply:
No, I put it in the header. Doing anything lavutil related throws "No such file or directory" errors in other tests, so I think it's not worth it.

Reply:
Weird, that doesn't sound right. I just tried it again (removed this line and added the include) and it was fine. Did you try a clean build?

Reply:
yea:
```
cld@kafka:~/fun/mpv$ rm -rf build
cld@kafka:~/fun/mpv$ git stash
Saved working directory and index state WIP on screenshot_avif: 05232151c5 test/test_utils: Add stub for mp_set_avdict
cld@kafka:~/fun/mpv$ meson setup -Dtests=true build
...
cld@kafka:~/fun/mpv$ meson compile -C build
[19/231] Compiling C object test/linked-list.p/linked_list.c.o
FAILED: test/linked-list.p/linked_list.c.o 
cc -Itest/linked-list.p -Itest -I../test -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -std=c11 -O2 -g -D_ISOC99_SOURCE -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=implicit-function-declaration -Wno-error=deprecated-declarations -Wno-error=unused-function -Wempty-body -Wdisabled-optimization -Wstrict-prototypes -Wno-format-zero-length -Wno-redundant-decls -Wvla -Wno-format-truncation -Wimplicit-fallthrough -fno-math-errno -Wformat -Werror=format-security -Wundef -Wmissing-prototypes -Wshadow -Wno-switch -Wparentheses -Wpointer-arith -Wno-pointer-sign -Wno-unused-result -MD -MQ test/linked-list.p/linked_list.c.o -MF test/linked-list.p/linked_list.c.o.d -o test/linked-list.p/linked_list.c.o -c ../test/linked_list.c
In file included from ../test/linked_list.c:3:
../test/test_utils.h:8:10: fatal error: libavutil/dict.h: No such file or directory
    8 | #include "libavutil/dict.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[20/231] Compiling C object test/gl-video.p/gl_video.c.o
FAILED: test/gl-video.p/gl_video.c.o 
cc -Itest/gl-video.p -Itest -I../test -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -std=c11 -O2 -g -D_ISOC99_SOURCE -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=implicit-function-declaration -Wno-error=deprecated-declarations -Wno-error=unused-function -Wempty-body -Wdisabled-optimization -Wstrict-prototypes -Wno-format-zero-length -Wno-redundant-decls -Wvla -Wno-format-truncation -Wimplicit-fallthrough -fno-math-errno -Wformat -Werror=format-security -Wundef -Wmissing-prototypes -Wshadow -Wno-switch -Wparentheses -Wpointer-arith -Wno-pointer-sign -Wno-unused-result -MD -MQ test/gl-video.p/gl_video.c.o -MF test/gl-video.p/gl_video.c.o.d -o test/gl-video.p/gl_video.c.o -c ../test/gl_video.c
In file included from ../test/gl_video.c:1:
../test/test_utils.h:8:10: fatal error: libavutil/dict.h: No such file or directory
    8 | #include "libavutil/dict.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[21/231] Compiling C object test/chmap.p/chmap.c.o
FAILED: test/chmap.p/chmap.c.o 
cc -Itest/chmap.p -Itest -I../test -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -std=c11 -O2 -g -D_ISOC99_SOURCE -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=implicit-function-declaration -Wno-error=deprecated-declarations -Wno-error=unused-function -Wempty-body -Wdisabled-optimization -Wstrict-prototypes -Wno-format-zero-length -Wno-redundant-decls -Wvla -Wno-format-truncation -Wimplicit-fallthrough -fno-math-errno -Wformat -Werror=format-security -Wundef -Wmissing-prototypes -Wshadow -Wno-switch -Wparentheses -Wpointer-arith -Wno-pointer-sign -Wno-unused-result -MD -MQ test/chmap.p/chmap.c.o -MF test/chmap.p/chmap.c.o.d -o test/chmap.p/chmap.c.o -c ../test/chmap.c
In file included from ../test/chmap.c:4:
../test/test_utils.h:8:10: fatal error: libavutil/dict.h: No such file or directory
    8 | #include "libavutil/dict.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[22/231] Compiling C object test/json.p/json.c.o
FAILED: test/json.p/json.c.o 
cc -Itest/json.p -Itest -I../test -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -std=c11 -O2 -g -D_ISOC99_SOURCE -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=implicit-function-declaration -Wno-error=deprecated-declarations -Wno-error=unused-function -Wempty-body -Wdisabled-optimization -Wstrict-prototypes -Wno-format-zero-length -Wno-redundant-decls -Wvla -Wno-format-truncation -Wimplicit-fallthrough -fno-math-errno -Wformat -Werror=format-security -Wundef -Wmissing-prototypes -Wshadow -Wno-switch -Wparentheses -Wpointer-arith -Wno-pointer-sign -Wno-unused-result -MD -MQ test/json.p/json.c.o -MF test/json.p/json.c.o.d -o test/json.p/json.c.o -c ../test/json.c
In file included from ../test/json.c:3:
../test/test_utils.h:8:10: fatal error: libavutil/dict.h: No such file or directory
    8 | #include "libavutil/dict.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[23/231] Compiling C object test/paths.p/paths.c.o
FAILED: test/paths.p/paths.c.o 
cc -Itest/paths.p -Itest -I../test -I. -I.. -fdiagnostics-color=always -Wall -Winvalid-pch -std=c11 -O2 -g -D_ISOC99_SOURCE -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64 -Werror=implicit-function-declaration -Wno-error=deprecated-declarations -Wno-error=unused-function -Wempty-body -Wdisabled-optimization -Wstrict-prototypes -Wno-format-zero-length -Wno-redundant-decls -Wvla -Wno-format-truncation -Wimplicit-fallthrough -fno-math-errno -Wformat -Werror=format-security -Wundef -Wmissing-prototypes -Wshadow -Wno-switch -Wparentheses -Wpointer-arith -Wno-pointer-sign -Wno-unused-result -MD -MQ test/paths.p/paths.c.o -MF test/paths.p/paths.c.o.d -o test/paths.p/paths.c.o -c ../test/paths.c
In file included from ../test/paths.c:5:
../test/test_utils.h:8:10: fatal error: libavutil/dict.h: No such file or directory
    8 | #include "libavutil/dict.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
[26/231] Generating generated/version.h with a custom command
ninja: build stopped: subcommand failed.
```

Reply:
(also failes with `<libavutil/dict.h>`)

Reply:
How is that even possible? mpv itself uses that same header.

Reply:
I have no idea.